Queue

Rea888 opened the pull request.

Rea888 committed:
changed default connection of queue to database

Rea888 committed:
created jobs table

Rea888 committed:
created ProcessHumidity job class

Rea888 committed:
Deleted unecessary command
Deleted empty row
Edited getNextUnprocessedHumidityModel to 2 function
Deleted unnecessary constructor
Deleted unnecessary space in api call
Created HumidityQueueCommand
Edited ProcessHumidity

Rea888 committed:
edited date if today to 1 month earlier
added commands to schedule
edited readme

KuveeTheBird commented:
*sigh* i love descriptive branch names, like... 'queue'...

KuveeTheBird commented on app/Console/Commands/HumidityQueueCommand.php:
```diff
@@ -0,0 +1,59 @@
+     *
+     * @var string
+     */
+    protected $description = 'Find the null dates in database and save the humidity in a queue';
```
this description is gramattically incoherent, or simply a lie :)

Rea888 replied:
I changed it i hope i is more accurate

KuveeTheBird commented on app/Console/Commands/HumidityQueueCommand.php:
```diff
@@ -0,0 +1,59 @@
+            $job = new ProcessHumidity($unprocessedHumidityModel);
+            dispatch($job);
+
+            // Provide feedback to the console
```
unnecessary comment, pls remove

KuveeTheBird commented on app/Console/Commands/HumidityQueueCommand.php:
```diff
@@ -0,0 +1,59 @@
+            $job = new ProcessHumidity($unprocessedHumidityModel);
+            dispatch($job);
+            $this->info(sprintf("ProcessHistoricalHumidity job (id: %s) dispatched to the queue.", $unprocessedHumidityModel->id));
+
```
remove empty lines, and apply the same formatting as in the foreach above

KuveeTheBird commented on app/Console/Kernel.php:
```diff
@@ -16,6 +16,11 @@ class Kernel extends ConsoleKernel
     protected function schedule(Schedule $schedule)
     {
         $schedule->command('weatherforecast:process')->dailyAt('01:00');
+        $schedule->command('humidityProcess:save')->monthly();
```
i think this way you'll experience 1 month delay in processing, unless humidityProcess starts AND finishes before humidity:queue even starts...
also wgy is the humidity:queue and queue:work lines duplicated?

Rea888 replied:
I changed $today to $todayMinusOneMonth because we asks one month periods and till is not end of the month the work gonna fail anyway,,, I run the command twice because at first it finds the nulls of began_at and it finish the job...When i run it for the second time it finds  the nulls of finsihed_at and finish the job

Rea888 replied:
I separated the two jobs in two different command so it has better fit to single responsibility principle and i dont repeat myself :)

KuveeTheBird commented on app/Http/Controllers/HistoricalWeatherHumidityController.php:
```diff
@@ -2,26 +2,13 @@
 use App\Repository\LongitudeLatitudeRepository;
 use App\Service\HistoricalWeatherHumidityService;
 
 class HistoricalWeatherHumidityController extends Controller
```
if this controller is empty , why keep it at all?
delete the file pls

KuveeTheBird commented on app/Jobs/ProcessHumidity.php:
```diff
@@ -0,0 +1,42 @@
+     */
+    public function __construct(HistoricalHumidityProcessingReportsModel $historicalHumidityProcessingReportsModel)
+    {
+
```
remove empty line

KuveeTheBird commented on app/Jobs/ProcessHumidity.php:
```diff
@@ -0,0 +1,42 @@
+
+    private HistoricalHumidityProcessingReportsModel $historicalHumidityProcessingReportsModel;
+
+    /**
```
remove this comment block

KuveeTheBird commented on app/Jobs/ProcessHumidity.php:
```diff
@@ -0,0 +1,42 @@
+        $this->historicalHumidityProcessingReportsModel = $historicalHumidityProcessingReportsModel;
+    }
+
+    /**
```
remove this comment block

KuveeTheBird commented on app/Service/HistoricalWeatherHumidityService.php:
```diff
@@ -28,7 +28,7 @@ public function getHistoricalHumidityFromApi($cityName, $year, $month): array
         $endDate = date("Y-m-t", strtotime($startDate));
 
-        $apiResponse = Http::get('https://archive-api.open-meteo.com/v1/archive?latitude=' . $latitude . '&longitude=' . $longitude . '&start_date=' . $startDate . '&end_date=' . $endDate . '&hourly=relativehumidity_2m');
+        $apiResponse = Http::get('https://archive-api.open-meteo.com/v1/archive?latitude='.$latitude.'&longitude='.$longitude.'&start_date='.$startDate.'&end_date='.$endDate.'&hourly=relativehumidity_2m');
```
you should use alt+enter and convert to sprintf() call

sprintf is a reaaaaaaly nice helper function, you should look it up

Rea888 committed:
Deleted HistoricalWeatherHumidityController.php
Converted to sprintf()
Edited description deleted empty lines unnecessary comments

Rea888 committed:
deleted unnecessary lines
deleted unnecessary lines
renamed and edited HumidityQueueCommand.php
created HumidityFinishQueueCommand.php
edited schedule()
deleted unnecessary import
